### ISSUE TITLE
Incorrect Analysis: Resistor Color Duo

### ISSUE THREAD

**Issue by Tyresius92** [exercism/javascript-analyzer#101]
**Describe the incorrectness**
A clear and concise description of what you think is incorrect.

I believe that the analyzer gave a false positive on an issue. It said that `colorCode` was not implemented optimally. Short of removing the curly braces and return keyword (which I think is a stylistic choice), I don't think it could be any better. 

**Which exercise**

Resistor Color Duo

**Source file(s)**

```javascript
// source file: resistor-color-duo.js

export const decodedValue = (colors) => {
  let indices = colors.map(colorCode)
  let result = "" + indices[0] + indices[1]
  return Number(result)
}
const colorCode = (color) => {
  return COLORS.indexOf(color)
}
export const COLORS = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "White"]
```

**Expected analysis**

I expected the analyzer not to flag this as incorrect. 

**Additional context**

**Comment by SleeplessByte** [comment]
Thanks!

`colorCode` is indeed optimally implemented, but `decodedValue` isn't. If the commentary complains about `colorCode`, then that's indeed a bug.

**Comment by SleeplessByte** [comment]
@Tyresius92 the problem is actually the `COLORS` array! It is not optimal, because it has `White` instead of `white`. If the `COLORS` array is fixed, the commentary becomes:

```json
{
  "comments": [
    {
      "comment": "javascript.resistor-color-duo.must_add_missing_call",
      "params": {
        "method.name": ".join('')",
        "expected.reason": "glue the digits together as a single number-string"
      },
      "type": "actionable"
    },
    {
      "comment": "javascript.resistor-color-duo.limit_number_of_colors",
      "type": "essential"
    }
  ]
}
```

Still not optimal messaging, but what it complains about is `.map()` followed by a string concatenation, which isn't great.
